hbase-server/src/main/java/org/apache/hadoop/hbase/tool/BulkLoadHFiles.java

```diff
@@ -84,6 +84,10 @@ public Path getFilePath() {
   Map<LoadQueueItem, ByteBuffer> bulkLoad(TableName tableName, Map<byte[], List<Path>> family2Files)
       throws TableNotFoundException, IOException;
 
+  /**
+   * Disable replication for this bulkload, if bulkload replication is configured.
+   */
+  void disableReplication();
```

Comment:
This is `Public` so we need to be careful what branch this lands in, right?

Also, since this is directly user facing, would a `isReplicationDisabled()` getter be a good idea?

Reply:
> Also, since this is directly user facing, would a isReplicationDisabled() getter be a good idea?

Done

> This is Public so we need to be careful what branch this lands in, right?

Right! For master, should be ok. For branch-2, maybe add a _default static BulkLoadHFiles createNoReplication(Configuration conf)_ whose implementation already takes care of disabling it, without having to define that abstract _disableReplication()_ method?

Reply:
> For branch-2, maybe add a default static BulkLoadHFiles createNoReplication(Configuration conf) whose implementation already takes care of disabling it, without having to define that abstract disableReplication() method?

Sounds like a plan!